java/bogdana_maksiuta/src/com/zoolatech/lecture4/tasks/_2/BinarySearchTree.java

```diff
@@ -0,0 +1,59 @@
+package com.zoolatech.lecture4.tasks._2;
+
+public class BinarySearchTree {
+    private Node rootNode;
+
+    static class Node {
```

Comment:
the class Node can be private, we don't want anyone outside of the tree class to know our internal implementation